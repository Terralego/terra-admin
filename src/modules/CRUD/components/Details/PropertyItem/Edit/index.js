import { connectAppProvider } from '../../../../../../components/AppProvider';

import Edit from './Edit';

import compose from '../../../../../../utils/compose';

const appProviderGetter = ({
  env: { modules: { CRUD: { settings } } },
}) => ({
  settingsEndpoint: settings,
});

export default compose(
  connectAppProvider(appProviderGetter),
)(Edit);
