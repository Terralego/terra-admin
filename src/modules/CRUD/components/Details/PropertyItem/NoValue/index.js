import { withTranslation } from 'react-i18next';
import compose from '../../../../../../utils/compose';

import NoValue from './NoValue';

export default compose(
  withTranslation(),
)(NoValue);
