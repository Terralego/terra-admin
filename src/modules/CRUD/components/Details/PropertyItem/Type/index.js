import { withNamespaces } from 'react-i18next';
import compose from '../../../../../../utils/compose';

import Type from './Type';

export default compose(
  withNamespaces(),
)(Type);
