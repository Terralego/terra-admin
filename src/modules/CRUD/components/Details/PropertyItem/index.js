import { connectAuthProvider } from '@terralego/core/modules/Auth';
import PropertyItem from './PropertyItem';

import compose from '../../../../../utils/compose';


const connectAuthProviderGetter = (({
  authenticated,
  user,
}) => {
  const permissions = authenticated ? user.permissions : [];
  return {
    canViewFeature: permissions.includes('can_view_feature'),
    canUpdateFeature: permissions.includes('can_change_feature'),
  };
});

export default compose(
  connectAuthProvider(connectAuthProviderGetter),
)(PropertyItem);
