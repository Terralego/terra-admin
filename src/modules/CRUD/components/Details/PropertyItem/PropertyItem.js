import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Type from './Type';
import Edit from './Edit';


const PropertyItem = ({
  canUpdateFeature,
  canViewFeature,
  value,
  ...props
}) => {
  if (!canViewFeature) {
    return null;
  }
  const { editedItem, name } = props;
  const isCurrentEditedItem = editedItem === name;
  return (
    <li className={classnames({
      'details__list-item': true,
      'details__list-item--active': isCurrentEditedItem,
      'details__list-item--inactive': !isCurrentEditedItem && editedItem !== '',
    })}
    >
      <span className="details__list-label">{value.title}</span>
      <div className="details__list-value">
        <Type {...value} />
      </div>
      {canUpdateFeature && <Edit {...value} {...props} />}
    </li>
  );
};

PropertyItem.propTypes = {
  canUpdateFeature: PropTypes.bool,
  canViewFeature: PropTypes.bool,
  editedItem: PropTypes.string,
  name: PropTypes.string,
  value: PropTypes.shape({
    title: PropTypes.string,
  }),
};

PropertyItem.defaultProps = {
  canUpdateFeature: false,
  canViewFeature: false,
  editedItem: '',
  name: undefined,
  value: {
    title: '',
  },
};

export default PropertyItem;
