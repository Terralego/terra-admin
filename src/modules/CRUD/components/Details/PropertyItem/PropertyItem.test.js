import React from 'react';
import renderer from 'react-test-renderer';

import PropertyItem from './PropertyItem';

jest.mock('./Edit', () => props => <div {...props}>Edit</div>);
jest.mock('./Type', () => props => <div {...props}>Type</div>);

const props = {
  canViewFeature: true,
  canUpdateFeature: true,
  value: {
    title: 'Foo',
  },
};

describe('should render correctly', () => {
  it('active item', () => {
    const tree = renderer.create((
      <PropertyItem
        {...props}
        name="bar"
        editedItem="bar"
      />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('inactive item', () => {
    const tree = renderer.create((
      <PropertyItem
        {...props}
        name="test"
        editedItem="bar"
      />
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
