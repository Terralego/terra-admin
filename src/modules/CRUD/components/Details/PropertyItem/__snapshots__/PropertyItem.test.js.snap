// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly active item 1`] = `
<li
  className="details__list-item details__list-item--active"
>
  <span
    className="details__list-label"
  >
    Foo
  </span>
  <div
    className="details__list-value"
  >
    <div
      title="Foo"
    >
      Type
    </div>
  </div>
  <div
    editedItem="bar"
    name="bar"
    title="Foo"
  >
    Edit
  </div>
</li>
`;

exports[`should render correctly inactive item 1`] = `
<li
  className="details__list-item details__list-item--inactive"
>
  <span
    className="details__list-label"
  >
    Foo
  </span>
  <div
    className="details__list-value"
  >
    <div
      title="Foo"
    >
      Type
    </div>
  </div>
  <div
    editedItem="bar"
    name="test"
    title="Foo"
  >
    Edit
  </div>
</li>
`;
